litestar/channels/_redis_xadd_expire.lua

```diff
@@ -0,0 +1,13 @@
+local data = ARGV[1]
+local limit = ARGV[2]
+local exp = ARGV[3]
+local maxlen_approx = ARGV[4]
+
+for i, key in ipairs(KEYS) do
+    if maxlen_approx == 1 then
+        redis.call("XADD", key, "MAXLEN", "~", limit, "*", "data", data, "channel", ARGV[i + 4])
+    else
+        redis.call("XADD", key, "MAXLEN", limit, "*", "data", data, "channel", ARGV[i + 4])
```

Comment:
Can someone more experienced with lua tell me if there's a better way to do this?
I tried using `unpack` to construct a part of the args, but Redis throws a fit then and only seems to accept that when *all* the args are unpacked, which defeats the purpose.